### PR TITLE
Chore/update tooling and code style

### PR DESCRIPTION
**Pull request by icazevedo**
#### What does this PR do? \*

This PR adds Slots options alongside Extension Points. It also removes the use of ramda and makes the PermissionChallenge code more clear by moving each query to its own separate hook.

The PR also updates the tooling, updates CSS Handles to use the new major, and fixes TS errors on the entire app.

**I recommend reviewing commit by commit to make sense of the changes**

#### How to test it? \*

Look for this component: 
Screen Shot 2021-02-18 at 15 43 51

The changes shouldn't make any difference to this app's behavior. I'll give you the credentials to login and see the prices on slack :)

[Workspace with the changes](https://icarovtex--b2bstoreqa.myvtex.com/)
[Master workspace (before the changes)](b2bstoreqa.myvtex.com/)

### CODE REVIEW

File: react/modules/session.ts [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+}
+
+export interface SessionUnauthorized {
+  type: 'Unauthorized'
+  message: string
+}
+
+export interface RenderSession {
+  sessionPromise: Promise<SessionPromise>
+}
```
Comment (igorbrasileiro) [inline react/modules/session.ts:21]: Lint here

File: react/modules/session.ts [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
+export interface SessionPromise {
+  response: Session | SessionUnauthorized
+}
+
+export type SessionWindow = Window & {
+  __RENDER_8_SESSION__?: { sessionPromise?: Promise<SessionPromise> }
+}
```
Comment (igorbrasileiro) [inline react/modules/session.ts:28]: Are these types necessary? `vtex setup --typings` shouldn't solve this?

**Review by igorbrasileiro — COMMENTED** [review]
Look comment on CHANGELOG and lint the files. LGTM

File: react/hooks/useSessionAuthorization.ts [hunk 3]
```diff
@@ -16,13 +17,11 @@ export const useSessionAuthorization = () => {
...
   }
 
-  return session &&
-  (session as SessionUnauthorized).type &&
-  (session as SessionUnauthorized).type.toLowerCase() === 'unauthorized'
-    ? false
-    : true
+  return !(
+    (session as SessionUnauthorized)?.type.toLowerCase() === 'unauthorized'
+  )
```
[the remark this answers is not in the document]
Comment (icazevedo) [reply react/hooks/useSessionAuthorization.ts:26]: 🤯 HAHAHAH very well noted!

File: react/hooks/useUserPermissionNames.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    return { userPermissionNames: null }
+  }
+
+  const userPermissionNames: string[] = permissionData.documents
+    .filter(document => permissionIds.includes(document.id))
+    .map(
+      document =>
+        document.fields?.find(field => field.key === 'name')?.value ?? ''
+    )
+    .filter(Boolean)
```
[the remark this answers is not in the document]
Comment (icazevedo) [reply react/hooks/useUserPermissionNames.ts:38]: I remember discussing this practice some time ago, I think we landed on `.filter(Boolean)` but I'm not sure. @victorhmp can you remember?

File: react/modules/session.ts (hunk 2 above)
Comment (icazevedo) [reply react/modules/session.ts:28]: They are 😞  The types provided by this command are mainly related to the `useRuntime` hook.

File: package.json [hunk 5]
```diff
@@ -17,6 +16,27 @@
...
+      "prettier --write"
+    ]
+  },
+  "devDependencies": {
+    "@vtex/intl-equalizer": "^2.3.0",
+    "@vtex/prettier-config": "^0.1.4",
+    "eslint": "^7.12.1",
+    "eslint-config-vtex": "^12.3.2",
+    "eslint-config-vtex-react": "^6.3.2",
+    "husky": "^4.2.3",
```
[the remark this answers is not in the document]
Comment (icazevedo) [reply package.json:34]: Sure!

**Review by igorbrasileiro — APPROVED** [review]
LGTM

File: react/hooks/useOrgAssignment.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import documentQuery from '../graphql/documents.graphql'
+import {
+  ORG_ASSIGNMENT_FIELDS,
+  ORG_ASSIGNMENT_SCHEMA,
+  ORG_ASSIGNMENT,
+} from '../utils/const'
+import type { MDSearchData } from '../PermissionChallenge'
+import type { UseProfileDataReturn } from './useProfileData'
+
+export interface UseOrgAssisgnmentReturn {
```
Comment (victorhmp) [inline react/hooks/useOrgAssignment.ts:12]:
```suggestion
export interface UseOrgAssignmentReturn {
```

File: react/hooks/useOrgAssignment.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  roleId?: string | null
+}
+
+export function useOrgAssignment({
+  email,
+  organizationId,
+}: UseProfileDataReturn): UseOrgAssisgnmentReturn {
```
Comment (victorhmp) [inline react/hooks/useOrgAssignment.ts:19]:
```suggestion
}: UseProfileDataReturn): UseOrgAssignmentReturn {
```

File: react/hooks/useUserPermissionNames.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (victorhmp) [reply react/hooks/useUserPermissionNames.ts:38]: Nope 😅 .
But I quite like it!

File: react/modules/session.ts [hunk 8]
```diff
@@ -1,11 +1,35 @@
-import { SessionPromise } from 'vtex.render-runtime'
+export interface KeyValue {
```
Comment (victorhmp) [inline react/modules/session.ts:1]: You might not need this anymore!
Try using `vtex.session-client@1.0.0`. I know you are just updating things here, so if it doesn't work straight away, I guess it's fine to keep this file. But it's worth a try ;)

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/modules/session.ts (hunk 8 above)
Comment (icazevedo) [reply react/modules/session.ts]: I've just found out that this file isn't being used at all 🤯 

Actually, it is, but the only hook that's using it isn't used so 🤷

File: react/PermissionChallenge.tsx [hunk 9]
```diff
@@ -89,40 +118,70 @@ function PermissionChallenge({ permissions = [] }: Props) {
...
+  const profileData = useProfileData()
+  const orgAssignment = useOrgAssignment(profileData)
+  const permissionIds = usePermissionIds(orgAssignment)
+  const { userPermissionNames } = useUserPermissionNames(permissionIds)
+
+  let hasPermission = false
+
+  if (userPermissionNames) {
+    hasPermission =
+      permissions.filter(permission =>
```
Comment (klzns) [inline react/PermissionChallenge.tsx:49]: Could use a `find` instead

**Review by klzns — APPROVED** [review]

**Pull request merged**